zebra-consensus/src/block/check.rs

```diff
@@ -25,3 +30,40 @@ pub fn is_coinbase_first(block: &Block) -> Result<(), Error> {
     }
     Ok(())
 }
+
+/// [3.9]: https://zips.z.cash/protocol/protocol.pdf#subsidyconcepts
+pub fn is_subsidy_correct(block: &Block) -> Result<(), Error> {
+    let height = block.coinbase_height().unwrap();
+
+    let coinbase = block.transactions.get(0).ok_or("no coinbase transaction")?;
+    let outputs = coinbase.outputs();
+
+    // Todo: we need the network here.
+    let network = Network::Mainnet;
```

Comment:
I think the right way to get the network here is to pass it as a parameter to `is_subsidy_correct`.  To pass it in, the block verifier will need to have the network, so we'll need to change the block verifier to hold onto the network it's supposed to be verifying.

The code that sets up the block verifier was completely changed in #1043, so making these changes should wait until after that PR lands (hopefully soon).

Reply:
#1043 is merged, could you try rebasing this branch onto `main`?

Reply:
Rebased.